src/mongoose_scram.erl

```diff
@@ -85,18 +85,6 @@ client_proof_key(ClientProof, ClientSignature) ->
 server_signature(Sha, ServerKey, AuthMessage) ->
     crypto:hmac(Sha, ServerKey, AuthMessage).
 
--spec hi(sha_type(), binary(), binary(), non_neg_integer()) -> binary().
-hi(Sha, Password, Salt, IterationCount) ->
-    U1 = crypto:hmac(Sha, Password, <<Salt/binary, 0, 0, 0, 1>>),
-    mask(U1, hi_round(Sha, Password, U1, IterationCount - 1)).
-
--spec hi_round(sha_type(), binary(), binary(), non_neg_integer()) -> binary().
-hi_round(Sha, Password, UPrev, 1) ->
-    crypto:hmac(Sha, Password, UPrev);
-hi_round(Sha, Password, UPrev, IterationCount) ->
-    U = crypto:hmac(Sha, Password, UPrev),
-    mask(U, hi_round(Sha, Password, U, IterationCount - 1)).
-
 -spec mask(binary(), binary()) -> binary().
 mask(Key, Data) ->
```

Comment:
By the way, I believe this `mask` function is equivalent to `crypto:exor/2`, so maybe we can also just use standard library functions and get rid of this one as well, it is used now in only one place. It even seems to have a very slightly better performance locally for me.